test/logger.test.js

```diff
@@ -301,6 +301,37 @@ describe('Logger (levels)', function () {
       .add(filterLevelTransport('ok'))
       .log(expected);
   });
+
+  it('sets transports levels', done => {
+    const logger = winston.createLogger();
+    const expectedError = { message: 'foo', level: 'error' };
+    const expectedInfo = { message: 'bar', level: 'info' };
+
+    function logLevelTransport(level) {
+      logger.level = level;
+      return new TransportStream({
+        log(obj) {
+          if (obj.level === 'info') {
+            assume(obj).equals(undefined, 'Transport on level info should never be called');
```

Comment:
@ChrisAlderson maybe I'm being dumb but if `obj` has `level` then how will it be `undefined` / do you actually enter this `if`?  LGTM aside from that question :)

Reply:
The `if`-statement might be a bit redundant because logs with `level = 'info'` should never go in the `log` method since they are being filtered in [`winston-transport` on L79](https://github.com/winstonjs/winston-transport/blob/9fd610af129b56253e62eedae6bc20f6fa06084b/index.js#L79). I added the `if`-statement because I followed the design of the other test cases in the test suite.

Reply:
Right.  Doesn't bother me much but maybe we could say `assume(obj.level).equals('error')` instead?  Maybe some of the other test cases in the suite could also be edited if they have dead code.  But anyway the fix works, that's the important thing :)

Reply:
It seems the `if`-statement is just another way to make the sure the test case fails if the implementation of the `Logger` is incorrect, so I wouldn't say it is dead code per se. You are right about `assume(obj.level).equals('error')`, which is already implemented, also makes the test fail. Guess we will see what @indexzero has to say. 

Thanks for the review 👍

Reply:
Fair point, you have my +1 :)

Reply:
This discussion is superseded now by @ChrisAlderson 's `refactor/tests` branch

Reply:
Agree with what you said @ChrisAlderson: 

> the `if`-statement is just another way to make the sure the test case fails if the implementation of the Logger is incorrect, so I wouldn't say it is dead code per se.

As I was refactoring that code I briefly broke something and that assertion was very helpful 👍